service/configinit.go

```diff
@@ -0,0 +1,546 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service // import "go.opentelemetry.io/collector/service"
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/AlecAivazis/survey/v2/terminal"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/provider/yamlprovider"
+	"go.opentelemetry.io/collector/service/internal/components"
+)
+
+const (
+	sampleConfigHeader = `Opentelemetry Collector sample configuration. Double check the pipeline configuration
+by ensuring the desired components belong to the respective pipeline type(ex metrics, logs, traces).`
+)
+
+type configCommand struct {
+	exiter    func(args ...error) error
+	stdio     terminal.Stdio
+	available *componentsAvailable
+}
+
+type componentsAvailable struct {
+	exporters  []string
+	processors []string
+	receivers  []string
+	extensions []string
+	pipelines  []string
+}
+
+type optionsSelected struct {
+	Exporters  []string
+	Processors []string
+	Receivers  []string
+	Extensions []string
+	Pipelines  []string
+
+	ConfigFileName string
+	DryRun         bool
+}
+
+var defaultConfigFileName = "otelcol.config.yaml"
+
+var cfgCmd *configCommand
+var log *zap.Logger
+var opts = new(optionsSelected)
+
+func init() {
+	var err error
+	log, err = zap.NewDevelopment(zap.AddStacktrace(zap.DPanicLevel))
+	if err != nil {
+		panic(fmt.Sprintf("failed to obtain a logger instance: %v", err))
+	}
+}
+
+// NewConfigurationInitCommand creates a new command for creating sample configurations
+func NewConfigurationInitCommand(set CollectorSettings) *cobra.Command {
+	cfgCmd = &configCommand{
+		exiter: func(args ...error) error {
+			log.Error(fmt.Sprintf("command failed %v", args))
+			return nil
+		},
+		stdio:     terminal.Stdio{In: os.Stdin, Out: os.Stdout, Err: os.Stderr},
+		available: getAvailableComponentsForPrompts(set.Factories),
+	}
+	opts = &optionsSelected{}
+	cmd := &cobra.Command{
+		Use:          "config",
+		Short:        "Generate a sample configuration file.",
+		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// no flags were passed(except dry run), show prompt
+			if (len(os.Args) == 3 && cmd.Flags().Lookup("dry-run").Changed) || len(os.Args) <= 2 {
+				err := showPrompt(cfgCmd.available, opts, cfgCmd.stdio)
+				// err could be interrupt or error occurred due to failed write to the answer object,
+				// in either case, we can't proceed further
+				if err != nil {
+					return cfgCmd.exiter(err)
+				}
+			}
+
+			exportersNode, err := getExportersNode(set.Factories, opts.Exporters)
```

Comment:
Could you split the business from the plumbing? Plumbing would be setting up the logger, validating the user input, returning errors, printing the outcome, and so on. Assembling the list of components and their sample config would be the business.